docs/adr/adr-008-p2p-discovery.md

```diff
@@ -0,0 +1,74 @@
+# ADR #008: Celestia-Node Full Node Discovery
+
+## Changelog
+
+- 2022.07.04 - discovery data structure
+
+## Authors
+
+@vgonkivs
+
+## Context
+
+This ADR is intended to describe p2p full node discovery in celestia node. 
+P2P discovery helps for light and full nodes to find other full nodes on the network at specified topic(`full`).
+As soon as full node is found and connection is established with it, then it will be added to a set of peers(limitedSet).
```

Comment:
```suggestion
P2P discovery helps light and full nodes to find other full nodes on the network at the specified topic(`full`).
As soon as a full node is found and connection is established with it, then it(full node) will be added to a set of peers(limitedSet).
```
```suggestion
P2P discovery helps for light and full nodes to find other full nodes on the network at specified topic(`full`).
As soon as full node is found and connection is established with it, then it will be added to a set of peers(limitedSet).
```
```suggestion
P2P discovery helps for light and full nodes to find other full nodes on the network at specified topic(`full`).
As soon as full node is found and connection is established with it, then it will be added to a set of peers(limitedSet).
```

Reply:
Please elaborate more on what is the `limitedSet`. Is it `peersLimit`? 🧐